dbms/programs/performance-test/PerformanceTest.cpp

```diff
@@ -296,6 +297,52 @@ void PerformanceTest::runQueries(
             break;
         }
     }
+
+    if (got_SIGINT)
+    {
+        return;
+    }
+
+    // Pull memory usage data from query log. The log entries take some time
+    // to become available, so we do this after running all the queries, and
+    // might still have to wait.
+    const int one_wait_us = 500 * 1000;
+    const int max_waits = (30 * 1000 * 1000) / one_wait_us;
+    int n_waits = 0;
+    for (auto & statistics : statistics_by_run)
+    {
+retry:
+        RemoteBlockInputStream log_reader(connection,
+            "select memory_usage from system.query_log where type = 2 and query_id = '"
+                                   + statistics.query_id + "'",
+            {} /* header */, context);
+
+        log_reader.readPrefix();
+        Block block = log_reader.read();
+        if (block.columns() == 0)
+        {
+            log_reader.readSuffix();
+
+            if (n_waits >= max_waits)
+                break;
+            n_waits++;
+
+            ::usleep(one_wait_us);
+            goto retry;
+        }
+        assert(block.columns() == 1);
+        assert(block.getDataTypes()[0]->getName() == "UInt64");
+        ColumnPtr column = block.getByPosition(0).column;
+        assert(column->size() == 1);
+        StringRef ref = column->getDataAt(0);
+        assert(ref.size == sizeof(UInt64));
+        statistics.memory_usage = *reinterpret_cast<const UInt64*>(ref.data);
+        log_reader.readSuffix();
+    }
+
+    // LOG_INFO can't format floats. Print to stderr because stdout is the
+    // resulting JSON.
+    fprintf(stderr, "Waited for query log for %.2fs\n", (n_waits * one_wait_us) / 1e6f);
```

Comment:
`LOG_INFO(log, "Waited for query log for " << std::setprecision(3) << (n_waits * one_wait_us) / 1e6f)` should be ok.